### PR TITLE
Migration guide for 0.3.0

### PR DESCRIPTION
**Pull request by Mr3zee**
**Subsystem**
Documentation

**Problem Description**
0.3.0 release contains breaking changes

**Solution**
Describe changes with a migration path

### CODE REVIEW

File: docs/pages/kotlinx-rpc/topics/0-3-0.topic [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+        SYSTEM "https://resources.jetbrains.com/writerside/1.0/xhtml-entities.dtd">
+<topic xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xsi:noNamespaceSchemaLocation="https://resources.jetbrains.com/writerside/1.0/topic.v2.xsd"
+       title="Migration to 0.3.0" id="0-3-0">
+
+    <p>
+        Version <code>0.3.0</code> introduces breaking changes.
+    </p>
+    <chapter title="Packages and artifacts changes" id="package-and-artifacts-changes">
+        This release addressed the issue of kRPC declarations being present in non-kRPC artifacts.
```
Comment (vnikolova) [inline docs/pages/kotlinx-rpc/topics/0-3-0.topic:12]:
```suggestion
        This release resolves the issue of kRPC declarations being incorrectly included in non-kRPC artifacts.
```

File: docs/pages/kotlinx-rpc/topics/0-3-0.topic [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+        At the same time, all package name were revisited and adjusted to match artifact names and the criteria of uniqueness.
```
Comment (vnikolova) [inline docs/pages/kotlinx-rpc/topics/0-3-0.topic:13]:
```suggestion
        All package names have been reviewed and updated to match artifact names and ensure uniqueness across the project.
```

**Review by vnikolova — APPROVED** [review]
Looks good, but some parts might be a bit confusing (in my comments).

File: docs/pages/kotlinx-rpc/topics/0-3-0.topic [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+        The table below contains the full list of the changes.
+
+        <table>
+            <tr>
+                <td><code>0.2.4</code></td>
+                <td><code>0.3.0</code></td>
+            </tr>
+            <tr>
+                <td><code>kotlinx.rpc.client.withService</code></td>
+                <td>Removed as deprecated</td>
```
[the remark this answers is not in the document]
Comment (Mr3zee) [reply docs/pages/kotlinx-rpc/topics/0-3-0.topic:24]: It was deprecated in 0.2.4 and removed in this release

File: docs/pages/kotlinx-rpc/topics/0-3-0.topic [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+                    <p>Artifact: <code>kotlinx-rpc-core</code></p>
+                </td>
+                <td>
+                    <code>kotlinx.rpc.krpc.streamScoped</code>
+                    <p>Artifact: <code>kotlinx-rpc-krpc-core</code></p>
+                </td>
+            </tr>
+            <tr>
+                <td>
+                    <code>kotlinx.rpc.client.KRPCClient</code>
```
[the remark this answers is not in the document]
Comment (Mr3zee) [reply docs/pages/kotlinx-rpc/topics/0-3-0.topic:116]: The ones that don't have the artifact field stayed in the same one.
I can make three table: removed, switch with artifact and without

**Review by e5l — APPROVED** [review]

**Pull request merged**